simply/market.py

```diff
@@ -228,21 +246,28 @@ def create_csv(self, filename, headers):
             writer = csv.writer(f)
             writer.writerow(headers)
 
-    def get_grid_fee(self, match):
+    def get_grid_fee(self, match=None, bid_cluster=None, ask_cluster=None):
         """
         Returns the grid fee associated with the bid and ask clusters of a given match.
 
         :param match: a dictionary representing a match, with keys 'bid_cluster' and 'ask_cluster'
+        :param ask_cluster: cluster id of bid
+        :param bid_cluster: cluster id of ask
         :return: the grid fee associated with the given bid and ask clusters
         """
+        if match is not None:
+            # if match is given, data from the match is used. In other cases bid
+            bid_cluster = match['bid_cluster']
+            ask_cluster = match['ask_cluster']
+
         if not self.grid_fee_matrix:
-            return 0
+            return cfg.config.default_grid_fee
         else:
-            if match['bid_cluster'] is None or match['ask_cluster'] is None:
+            if bid_cluster is None or ask_cluster is None:
```

Comment:
trigger warning, that at least one cluster is not defined, and that default grid fee is returned.

also, this case does not seem to be tested, as it would not work, e.g. if `self.grid_fee_matrix = None`, then `return self.grid_fee_matrix[0][-1]` would lead to an error

Reply:
`self.grid_fee_matrix = None`
would return right before that with the default_grid_fee

Reply:
I inverted the expression twice for some reason ^^ Seems like the afternoon slump fooled me, you are right.

However, first part about the warning still holds; making clear, that in case only one cluster was None the `self.grid_fee_matrix[0][-1]` default fee is returned

Reply:
added a warning

Reply:
the warning was there when I reviewed it but the statement is unclear:
e.g. `warnings.warn("At least one cluster is 'None', returning default grid fee.")`

Reply:
changed it to your suggestion